### ISSUE TITLE
Illegal instruction

### ISSUE THREAD

**Issue by jorvis** [torognes/vsearch#187]
I downloaded vsearch-2.0.2-linux-x86_64 and when I try to run it I get this:

```
$ ./bin/vsearch 
Illegal instruction

$ uname -a
Linux bfx 2.6.32-504.16.2.el6.x86_64 #1 SMP Wed Apr 22 06:48:29 UTC 2015 x86_64 x86_64 x86_64 GNU/Linux

$ lsb_release -a
LSB Version:	:base-4.0-amd64:base-4.0-noarch:core-4.0-amd64:core-4.0-noarch:graphics-4.0-amd64:graphics-4.0-noarch:printing-4.0-amd64:printing-4.0-noarch
Distributor ID:	CentOS
Description:	CentOS release 6.7 (Final)
Release:	6.7
Codename:	Final
```

**Comment by jorvis** [comment]
As an update, I was able to build it from source and execute it successfully.

**Comment by colinbrislawn** [comment]
Does the bioconda package work for you or does it throw the same error? I'm curious to see why some builds work while others don't. 
https://anaconda.org/bioconda/vsearch

**Comment by xflouris** [comment]
@torognes : I've cross-checked it with Debian/Ubuntu/CentOS. Compiling from source works on all systems, but using the pre-compiled binaries fails on Ubuntu/CentOS.

Debian: works fine (below uname -a)
Linux callisto 3.16.0-4-amd64 #1 SMP Debian 3.16.7-ckt25-2+deb8u3 (2016-07-02) x86_64 GNU/Linux

Ubuntu: gives Illegal instruction
Linux hitssv109 2.6.32-33-server #72-Ubuntu SMP Fri Jul 29 21:21:55 UTC 2011 x86_64 x86_64 x86_64 GNU/Linux

Centos: gives illegal instruction
Linux hitssv535.h-its.org 3.10.0-327.22.2.el7.x86_64 #1 SMP Thu Jun 23 17:05:11 UTC 2016 x86_64 x86_64 x86_64 GNU/Linux

Checking the generated coredump, I see that the problem is in the `fillheader()` function. I suppose the problem might be in arch_get_memtotal(). In the pre-compiled vsearch, the pre-processor already (presumably) chose to do use the linux part, whereas CentOS should have the __APPLE__ part I think. I'm surprised it fails on Ubuntu though.

**Comment by torognes** [comment]
Strange... The binary is is compiled on Centos!

```
-bash-4.1$ uname -a
Linux login-0-0.local 2.6.32-573.18.1.el6.x86_64 #1 SMP Tue Feb 9 22:46:17 UTC 2016 x86_64 x86_64 x86_64 GNU/Linux
-bash-4.1$ lsb_release -a
LSB Version:	:base-4.0-amd64:base-4.0-noarch:core-4.0-amd64:core-4.0-noarch:graphics-4.0-amd64:graphics-4.0-noarch:printing-4.0-amd64:printing-4.0-noarch
Distributor ID:	CentOS
Description:	CentOS release 6.7 (Final)
Release:	6.7
Codename:	Final
```

It might have been compiled with an old gcc version though. I'll try a newer one.

**Comment by torognes** [comment]
Thank you for reporting this problem. The problem was due to bad compiler flags (e.g. -mavx) on Linux. Fixed in version 2.0.3 just released.

**Comment by bioteksampath** [comment]
Hi,
I having the same problem.
Installed and tried with anaconda2 and 3 - but both throws an error - Illegal instruction

[p2s@c2 MITE-Tracker]$ ~/anaconda3/pkgs/vsearch-2.9.0-h96824bc_0/bin/vsearch
Illegal instruction

[p2s@c2 MITE-Tracker]$ ~/anaconda2/pkgs/vsearch-2.9.0-h96824bc_0/bin/vsearch
Illegal instruction

$ uname -a
Linux carinata.agr.gc.ca 2.6.32-642.4.2.el6.x86_64 #1 SMP Tue Aug 23 19:58:13 UTC 2016 x86_64 x86_64 x86_64 GNU/Linux

your help will be much appriciatable

Thanks
sam

**Comment by torognes** [comment]
This problem has been solved in VSEARCH version 2.9.1 just released. Please download the new version. Sorry for the inconvenience.
